Source/SpacePort/SpacePort/Spaceport.cs

```diff
@@ -1,84 +1,219 @@
-            for(int i = 0; i < keys.Count; i++)
-            {
-                int amount = randomAmount.Next(0, 10);
-                string key = keys[i];
-                this.ShipTypes[key] = amount;
-            }
+            if (shipLength >= 1 && shipLength <= 499)
+                lotCost = 100;
 
-            parkingCapacity = ShipTypes;
-        }
+            else if (shipLength >= 500 && shipLength <= 999)
+                lotCost = 200;
 
-        public bool CheckParkingAvalibility(string shipType)
-        {
-            int value = ShipTypes[shipType];
-            return value > 0;            
-        }
+            else if (shipLength >= 1000 && shipLength <= 3499)
+                lotCost = 350;
 
-        public bool CheckMaxLots (string shipType)
-        {
-            int value = ShipTypes[shipType];
-            int maxValue = parkingCapacity[shipType];
-            return value <= maxValue;
-        }
+            else if (shipLength >= 3500 && shipLength <= 4999)
+                lotCost = 450;
 
-        public void Park(string shipType)
-        {
-            ShipTypes[shipType] -= 1;
+            else if (shipLength >= 5000 && shipLength <= 9999)
+                lotCost = 750;
+
+            else if (shipLength > 10000)
+                lotCost = 2500;
+            
+            totalCost = (int)Math.Ceiling(lotCost * duration);
+
+            return totalCost;
         }
-        
-        public void Leave(string shipType)
+
+        public int CalculateParkingAvailability(double shipLength)
         {
-            ShipTypes[shipType] += 1;
+            int lotsAvailable = 0;
+
+            if (shipLength >= 1 && shipLength <= 499)
+                lotsAvailable = parkingLots["Micro"];
+
+            else if (shipLength >= 500 && shipLength <= 999)
+                lotsAvailable = parkingLots["Small"];
+
+            else if (shipLength >= 1000 && shipLength <= 3499)
+                lotsAvailable = parkingLots["Medium"];
+
+            else if (shipLength >= 3500 && shipLength <= 4999)
+                lotsAvailable = parkingLots["Large"];
+
+            else if (shipLength >= 5000 && shipLength <= 9999)
+                lotsAvailable = parkingLots["Largest"];
+
+            else if (shipLength > 10000)
+                lotsAvailable = parkingLots["Mega"];
+
+            return lotsAvailable;
         }
 
-        public int CalculateParkingTariff(string shipType)
+        public void Dock(double shipLength)
         {
+            if (shipLength >= 1 && shipLength <= 499)
+                parkingLots["Micro"] -=1;
 
+            else if (shipLength >= 500 && shipLength <= 999)
+                parkingLots["Small"] -=1;
 
+            else if (shipLength >= 1000 && shipLength <= 3499)
+                parkingLots["Medium"] -=1;
 
-            //tmp value
-            return 10;
-        }
+            else if (shipLength >= 3500 && shipLength <= 4999)
+                parkingLots["Large"] -=1;
```

Comment:
Nothing big at all: But does `parkingLots["Large"]--;` work as well?

Reply:
yes it probobly will guess its a smaksak, I sometimes like it better with -=1, especially in methods, while I usually use -- ni for-loops e.g :)